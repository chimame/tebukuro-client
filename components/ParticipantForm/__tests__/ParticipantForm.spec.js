import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import { ParticipantForm } from '../index'

const testProps = {
  eventId: 1,
  hasEventWaitlist: true,
  isUserRegistered: false,
  message: null,
  onSubmit: () => {},
  onCancel: () => {},
}

const completeProps = {
  ...testProps,
  message: '参加登録が完了しました。',
}

describe('ParticipantForm', () => {
  describe('renders participant form component', () => {
    describe('when user has already registered for the event', () => {
      it('renders CancelRegistrationButton component.', () => {
        const wrapper =
          shallow(<ParticipantForm {...testProps} isUserRegistered />)
        const tree = shallowToJson(wrapper)

        expect(tree).toMatchSnapshot()
      })
    })

    describe('when user has NOT registered for the event yet', () => {
      it('renders ParticipantButton component.', () => {
        const wrapper =
          shallow(<ParticipantForm {...testProps} />)
        const tree = shallowToJson(wrapper)

        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('on completing registeration for event', () => {
    it('renders completion message', () => {
      const wrapper = shallow(<ParticipantForm {...completeProps} />)
      const tree = shallowToJson(wrapper)

      expect(tree).toMatchSnapshot()
    })
  })
})
