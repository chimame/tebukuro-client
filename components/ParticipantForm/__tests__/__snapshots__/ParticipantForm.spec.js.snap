// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantForm on completing registeration for event renders completion message 1`] = `
<div>
  <div>
    <p>
       
      参加登録が完了しました。
       
    </p>
    <ParticipantButton
      eventId={1}
      hasEventWaitlist={true}
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`ParticipantForm renders participant form component when user NOT signed in renders nothig. 1`] = `<div />`;

exports[`ParticipantForm renders participant form component when user has NOT registered for the event yet renders ParticipantButton component. 1`] = `
<div>
  <div>
    <p>
       
       
    </p>
    <ParticipantButton
      eventId={1}
      hasEventWaitlist={true}
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`ParticipantForm renders participant form component when user has already registered for the event renders CancelRegistrationButton component. 1`] = `
<div>
  <div>
    <p>
       
       
    </p>
    <CancelRegistrationButton
      eventId={1}
      onClick={[Function]}
    />
  </div>
</div>
`;
